contracts/rewards/DevGuildRewardsDistributor.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.19;
+
+import {IPassportUtils} from "../utils/IPassportUtils.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+ *        ---------::::
+ *     ---------:---::::::
+ *   -----------::---:::::::
+ *  ------------:.:--::::::::
+ * -------------: .:--::::::::
+ * -------------:   .:::::::::
+ * -------------:.......::::::
+ * -----:..    .:-------::::::
+ * --------:.. .:-------::::::
+ * ----------:..:--------:::::
+ *  -----------.:--------::::
+ *   ----------::--------:::
+ *     -------------------
+ *        -------------
+ *
+ *         Nation3 DAO
+ *     https://nation3.org
+ */
+contract DevGuildRewardsDistributor {
+    string public constant VERSION = "0.6.5";
+
+    address public owner;
+    IPassportUtils public passportUtils;
+    IERC20 public rewardToken;
+
+    mapping(address => uint256) public claimable;
+    uint256 public claimableTotal;
+
+    mapping(address => uint256) public claimed;
+    uint256 public claimedTotal;
+
+    error NotPassportOwner(address illegalAlien);
+    error PassportExpired(address citizen);
+
+    constructor(address passportUtils_, address rewardToken_) {
+        owner = address(msg.sender);
+        passportUtils = IPassportUtils(passportUtils_);
+        rewardToken = IERC20(rewardToken_);
+    }
+
+    function setOwner(address owner_) public {
+        require(msg.sender == owner, "You are not the owner");
+        owner = owner_;
+    }
+
+    function setPassportUtils(address passportUtils_) public {
+        require(msg.sender == owner, "You are not the owner");
+        passportUtils = IPassportUtils(passportUtils_);
+    }
+
+    function addReward(address citizen, uint256 amount) public {
+        require(msg.sender == owner, "You are not the owner");
+        claimable[citizen] += amount;
+        claimableTotal += amount;
+    }
+
+    function claim() public {
+        if (!passportUtils.isOwner(msg.sender)) {
+            revert NotPassportOwner(msg.sender);
+        }
+        if (passportUtils.isExpired(msg.sender)) {
+            revert PassportExpired(msg.sender);
+        }
+
+        uint256 claimableAmount = claimable[msg.sender];
+        rewardToken.transfer(msg.sender, claimableAmount);
+
+        claimable[msg.sender] = 0;
+        claimableTotal -= claimableAmount;
+
+        claimed[msg.sender] += claimableAmount;
+        claimedTotal += claimableAmount;
+    }
+}
```

Comment:
Nice, is there already a task for adding events? Events would be especially cool so that we can see some of this stuff on the graph.

Reply:
Added an issue for it here:  https://github.com/nation3/nationcred-contracts/issues/94